File: nodejs/number-complement-476.js
'use strict';
/**
 * @param {number} num
 * @return {number}
 */
var findComplement = function(num) {    
    let length = Math.floor(1+Math.log2(num));
    return ((1<<length)-1) ^ num;
};
console.log(findComplement(1<<5));

File: golang/numbercomplement476/number-complement-476.go
package main

import (
	"fmt"
	"math"
)

func main() {

	fmt.Println("Hello World")
	fmt.Println(findComplement(4))
}

func findComplement(num int) int {
	length := math.Floor(1 + math.Log2(float64(num)))
	return ((1 << uint(length)) - 1) ^ num
}
